src/header-utilities/exponentialSearch.hpp

```diff
@@ -0,0 +1,51 @@
+class ExponentialSearch {
+  template< typename RandomAccessIterator,
+	    typename Value,
+	    typename Compare >
+  static RandomAccessIterator apply( RandomAccessIterator left,
+				     RandomAccessIterator right,
+				     RandomAccessIterator end,
+				     Value&& value,
+				     Compare&& compare ){
```

Comment:
What is `[Cc]ompare`? I assume it is a function used to determine whether two iterators are equal, right?